src/settings.cpp

```diff
@@ -448,13 +448,26 @@ static void Write_ValidateSetting(void *ptr, const SettingDesc *sd, int32 val)
 			case SLE_VAR_U16:
 			case SLE_VAR_I32: {
 				/* Override the minimum value. No value below sdb->min, except special value 0 */
-				if (!(sdb->flags & SGF_0ISDISABLED) || val != 0) val = Clamp(val, sdb->min, sdb->max);
+				if (!(sdb->flags & SGF_0ISDISABLED) || val != 0) {
+					if (!(sdb->flags & SGF_MULTISTRING)) {
+						val = Clamp(val, sdb->min, sdb->max);
+					} else if (val < sdb->min || val > (int32)sdb->max) {
+						val = (int32)(size_t)sdb->def;
+					}
+				}
 				break;
```

Comment:
There is a WriteValue in the case below, but not in this one. Why?

Reply:
break vs return